### PR TITLE
Add LM training/decoding in swbd recipe

### PR DESCRIPTION
**Pull request by YosukeHiguchi**
LM can be trained with Fisher English training speech transcripts.
I'll also update fisher_swbd recipe based on this pull request.

### CODE REVIEW

**Comment by sw005320** [comment]
LGTM.
Can you also prepare a pre-trained model?
See https://github.com/espnet/espnet/blob/63b70198c411db61915aa51fa9dabbdc05646234/egs/librispeech/asr1/RESULTS.md#pytorch-large-transformer-with-specaug-4-gpus--large-lm

Also, this model does not include a spm model, which is important to reproduce the result. So, please extend `pack_model.sh` to include the spm model.

File: egs/swbd/asr1/run.sh [hunk 1]
```diff
@@ -22,16 +22,25 @@ do_delta=false
...
 recog_model=model.acc.best # set a model to be used for decoding: 'model.acc.best' or 'model.loss.best'
 n_average=10
+lang_model=rnnlm.model.best # set a language model to be used for decoding
 
 # data
 swbd1_dir=/export/corpora3/LDC/LDC97S62
 eval2000_dir="/export/corpora2/LDC/LDC2002S09/hub5e_00 /export/corpora2/LDC/LDC2002T43"
 rt03_dir=/export/corpora/LDC/LDC2007S10
+# path to the Fisher corpus LDC2004T19 LDC2005T19 LDC2004S13 LDC2005S13 for LM training (optional)
+fisher_dir=""
```
Comment (sw005320) [inline egs/swbd/asr1/run.sh]: you may add the CLSP fisher directory.

File: egs/swbd/asr1/run.sh [hunk 2]
```diff
@@ -174,6 +190,44 @@ if [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ]; then
...
+mkdir -p ${lmexpdir}
+
+if [ ${stage} -le 3 ] && [ ${stop_stage} -ge 3 ]; then
+    echo "stage 3: LM Preparation"
+    lmdatadir=data/local/lm_train_${bpemode}${nbpe}
+    mkdir -p data/local/lm_train ${lmdatadir}
+    cut -f 2- -d" " data/${train_set}/text | gzip -c > data/local/lm_train/${train_set}_text.gz
+    if [ -n "${fisher_dir}" ]; then
+        cut -f 2- -d" " data/train_fisher/text | gzip -c > data/local/lm_train/train_fisher_text.gz
+        zcat data/local/lm_train/${train_set}_text.gz data/local/lm_train/train_fisher_text.gz |\
```
Comment (sw005320) [inline egs/swbd/asr1/run.sh]: did you also add the swbd text data?
You seem to only use fisher text data.

Comment (YosukeHiguchi) [reply egs/swbd/asr1/run.sh]: swbd and fisher texts are concatenated and encoded at l208, 209.

**Comment by kan-bayashi** [comment]
I think we can specify lang_* dicrectory for —dict to pack spm without any modification.

**Comment by YosukeHiguchi** [comment]
> I think we can specify lang_* dicrectory for —dict to pack spm without any modification.

File: egs/swbd/asr1/run.sh (hunk 2 above)
Comment (sw005320) [reply egs/swbd/asr1/run.sh]: oh, I see.
Then, can you add a comment about it?

**Comment by sw005320** [comment]
thanks!
swbd results are reaching SOTA now.

**Pull request merged**